### ISSUE TITLE
Update the Rendering Functionality

### ISSUE THREAD

**Issue by BenjamenMeyer** [vegastrike/Vega-Strike-Engine-Source#170]
We have had a few issues floating around discussing the engine and rendering. This is a story to replace those and center the discussion around simply updating the rendering functionality.

Closes: #46 #72 

Discussion: We need to update the rendering functionality in order to take advantage of newer APIs (f.e Vulkan). There's some nice libraries out there we can use (see discussion in #72) and we have a few options:
- [ ] Direct [Vulkan](https://www.khronos.org/vulkan/) integration instead of OpenGL.
- [ ] Use [BGFX](https://github.com/bkaradzic/bgfx).
- [ ] Use [OpenSceneGraph](http://www.openscenegraph.org/).
    - [ ] Should look at [VulkanSceneGraph](https://github.com/vsg-dev) too as it is replacing OpenSceneGraph.
- [ ] Use [DiligentEngine](https://github.com/DiligentGraphics/DiligentEngine)

OpenGL will continue to work, but is not as efficient as Vulkan. At a minimum we should move to using Vulkan; however, using a library like BGFX or OpenSceneGraph will give us the ability to use OpenGL, Vulkan, and even other APIs where available (Mac, Windows).

**Comment by Loki1950** [comment]
Yes we do need to update our rendering pipeline our current shaders are not working properly several units miss a texture layer because the relevant shader fails to compile.Using Vulkan should give us some future proofing as there is talk of some OS's dropping OpenGL support.

**Comment by evertvorster** [comment]
There are a bunch of issues with the rendering of VegaStrike, especially with multiple high resolution monitors and Wayland. 
When considering various options, we should have the requirement that you should be able to pick which monitor the game will be displayed on, as well as the resolution. 

Currently VegaStrike in windowed mode shows up on my leftmost monitor, no matter if it is set to primary or not, and in fullscreen mode it does not rescale the monitor, so that a low resolution does not fill the entire monitor. 

This, unfortunately has a huge impact on the initial impression that VegaStrike creates
